wallet/src/de/schildbach/wallet/ui/coinbase/CoinBaseWebClientActivity.kt

```diff
@@ -0,0 +1,108 @@
+package de.schildbach.wallet.ui.coinbase
+
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.MenuItem
+import android.webkit.*
+import androidx.appcompat.widget.Toolbar
+import de.schildbach.wallet_test.R
+import org.dash.wallet.common.InteractionAwareActivity
+import org.dash.wallet.integration.liquid.databinding.ActivityLoginWebviewBinding
+
+class CoinBaseWebClientActivity : InteractionAwareActivity() {
+
+    private lateinit var binding: ActivityLoginWebviewBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginWebviewBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        initUI()
+    }
+    @SuppressLint("SetJavaScriptEnabled")
+    private fun initUI() {
+        val toolbar = findViewById<Toolbar>(org.dash.wallet.integration.liquid.R.id.toolbar)
+        setSupportActionBar(toolbar)
+        val actionBar = supportActionBar
+        actionBar?.apply {
+            setDisplayHomeAsUpEnabled(true)
+            setDisplayShowHomeEnabled(true)
+        }
+        setTitle(R.string.CoinBase)
+        // Clear all the Application Cache, Web SQL Database and the HTML5 Web Storage
+        WebStorage.getInstance().deleteAllData()
+
+        // Clear all the cookies
+        CookieManager.getInstance().removeAllCookies(null)
+        CookieManager.getInstance().flush()
+
+        binding.webView.clearCache(true)
+        binding.webView.clearFormData()
+        binding.webView.clearHistory()
+        binding.webView.clearSslPreferences()
+
+        binding.webView.webViewClient = MyWebViewClient()
+        binding.webView.webChromeClient = WebChromeClient()
+        binding.webView.settings.javaScriptEnabled = true
+        binding.webView.settings.allowFileAccess = false
+        WebStorage.getInstance().deleteAllData()
+
+        val loginUrl =
+            "https://www.coinbase.com/oauth/authorize?client_id=1ca2946d789bf6d986f26df03f4a52a8c6f1fe80e469eb1d3477e7c90768559a&redirect_uri=https://coin.base.test/callback&response_type=code&account=all&scope=wallet:accounts:read,wallet:user:read,wallet:transactions:transfer"
+
+        binding.webView.loadUrl(loginUrl)
```

Comment:
We are using a webview rather than a custom tab?  Uphold and Liquid use a CustomTab.  Not sure which is better.

What happens when the autologout is triggered?  It appears that this Activity will finish, since it is derived from `InteractionAwareActivity` rather than `LockScreenActivity`

Reply:
Derive `CoinBaseWebClientActivity` from `LockScreenActivity`